File: ts/project/.sdk/tm/kotlin/core/Point.kt
package KOTLINPACKAGE.core

import KOTLINPACKAGE.utility.struct.Struct

/** A single endpoint definition (typed view over a point map). */
@Suppress("UNCHECKED_CAST")
class Point(pointmap: Map<String, Any?>?) {

  var args: MutableMap<String, Any?>
  var rename: MutableMap<String, Any?>
  var method: String = ""
  var orig: String = ""
  var parts: MutableList<Any?> = mutableListOf()
  var params: MutableList<Any?>? = null
  var select: MutableMap<String, Any?>? = null
  var active: Boolean = false
  var relations: MutableList<Any?>? = null
  var alias: MutableMap<String, Any?> = linkedMapOf()
  var transform: MutableMap<String, Any?> = linkedMapOf()

  init {
    var v = Struct.getprop(pointmap, "args")
    if (v is MutableMap<*, *>) {
      this.args = v as MutableMap<String, Any?>
    } else {
      this.args = linkedMapOf()
      this.args["params"] = mutableListOf<Any?>()
    }

    v = Struct.getprop(pointmap, "rename")
    if (v is MutableMap<*, *>) {
      this.rename = v as MutableMap<String, Any?>
    } else {
      this.rename = linkedMapOf()
      this.rename["params"] = linkedMapOf<String, Any?>()
    }

    v = Struct.getprop(pointmap, "method")
    if (v is String) {
      this.method = v
    }

    v = Struct.getprop(pointmap, "orig")
    if (v is String) {
      this.orig = v
    }

    v = Struct.getprop(pointmap, "parts")
    if (v is MutableList<*>) {
      this.parts = v as MutableList<Any?>
    }

    v = Struct.getprop(pointmap, "params")
    if (v is MutableList<*>) {
      this.params = v as MutableList<Any?>
    }

    v = Struct.getprop(pointmap, "select")
    if (v is MutableMap<*, *>) {
      this.select = v as MutableMap<String, Any?>
    }

    v = Struct.getprop(pointmap, "active")
    if (v is Boolean) {
      this.active = v
    }

    v = Struct.getprop(pointmap, "relations")
    if (v is MutableList<*>) {
      this.relations = v as MutableList<Any?>
    }

    v = Struct.getprop(pointmap, "alias")
    if (v is MutableMap<*, *>) {
      this.alias = v as MutableMap<String, Any?>
    }

    v = Struct.getprop(pointmap, "transform")
    if (v is MutableMap<*, *>) {
      this.transform = v as MutableMap<String, Any?>
    }
  }
}

File: ts/project/.sdk/tm/csharp/core/Point.cs
// ProjectName SDK - typed view over an endpoint (point) definition.

using Voxgig.Struct;

namespace ProjectNameSdk;

public class Point
{
    public Dictionary<string, object?> Args = new() { ["params"] = new List<object?>() };
    public Dictionary<string, object?> Rename = new() { ["params"] = new Dictionary<string, object?>() };
    public string Method = "";
    public string Orig = "";
    public List<object?> Parts = new();
    public List<object?>? Params;
    public Dictionary<string, object?>? Select;
    public bool Active;
    public List<object?>? Relations;
    public Dictionary<string, object?> Alias = new();
    public Dictionary<string, object?> Transform = new();

    public Point(Dictionary<string, object?>? pointmap)
    {
        if (StructUtils.GetProp(pointmap, "args") is Dictionary<string, object?> am)
        {
            Args = am;
        }
        if (StructUtils.GetProp(pointmap, "rename") is Dictionary<string, object?> rm)
        {
            Rename = rm;
        }
        if (StructUtils.GetProp(pointmap, "method") is string m)
        {
            Method = m;
        }
        if (StructUtils.GetProp(pointmap, "orig") is string o)
        {
            Orig = o;
        }
        if (StructUtils.GetProp(pointmap, "parts") is List<object?> pl)
        {
            Parts = pl;
        }
        if (StructUtils.GetProp(pointmap, "params") is List<object?> pr)
        {
            Params = pr;
        }
        if (StructUtils.GetProp(pointmap, "select") is Dictionary<string, object?> sm)
        {
            Select = sm;
        }
        if (StructUtils.GetProp(pointmap, "active") is bool ab)
        {
            Active = ab;
        }
        if (StructUtils.GetProp(pointmap, "relations") is List<object?> rl)
        {
            Relations = rl;
        }
        if (StructUtils.GetProp(pointmap, "alias") is Dictionary<string, object?> al)
        {
            Alias = al;
        }
        if (StructUtils.GetProp(pointmap, "transform") is Dictionary<string, object?> tf)
        {
            Transform = tf;
        }
    }
}
